wrappers/react/src/hotTableInner.tsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import Handsontable from 'handsontable/base';
+import { SettingsMapper } from './settingsMapper';
+import { RenderersPortalManager } from './renderersPortalManager';
+import { isHotColumn } from './hotColumn';
+import { HotEditorHooks, HotTableProps, HotTableRef } from './types';
+import {
+  HOT_DESTROYED_WARNING,
+  AUTOSIZE_WARNING,
+  createEditorPortal,
+  getContainerAttributesProps,
+  isCSR,
+  warn,
+  displayObsoleteRenderersEditorsWarning,
+  useUpdateEffect
+} from './helpers';
+import PropTypes from 'prop-types';
+import { getRenderer } from 'handsontable/renderers/registry';
+import { getEditor } from 'handsontable/editors/registry';
+import { useHotTableContext } from './hotTableContext'
+import { HotColumnContextProvider } from './hotColumnContext'
+import { EditorContextProvider, makeEditorClass } from "./hotEditor";
+
+const HotTableInner = React.forwardRef<HotTableRef, HotTableProps>((props, ref) => {
```

Comment:
I initially planned to merge this (what used to be `HotTableClass` into `HotTable` itself, but it's easier to use context if we keep this separation - `HotTable` is the context provider while `HotTableInner` is its consumer. I'm open for the better inner component name suggestion.

Reply:
Your justification for having a distinct `hotTable.tsx` and `hotTableInner.tsx` makes sense to me, I wouldn't worry about changing the `HotTableInner` as its pretty self-explanatory and not exposed to the end user.